### PR TITLE
LG queries can be undefined

### PR DESCRIPTION
**Pull request by bruugey**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [Name of ticket](https://jira.mongodb.org/browse/[name-of-ticket])

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: scripts/tsconfig.json [hunk 1]
```diff
@@ -17,5 +17,6 @@
     "baseUrl": ".",
     "skipLibCheck": true,
     "resolveJsonModule": true,
+    "noImplicitAny": false
```
Comment (bruugey) [inline scripts/tsconfig.json:20]: this can be rm'd

File: tools/build/tsconfig.json [hunk 2]
```diff
@@ -4,7 +4,8 @@
     "declarationDir": "dist",
     "outDir": "dist",
     "rootDir": "src",
-    "baseUrl": "."
+    "baseUrl": ".",
+    "noImplicitAny": false
```
Comment (bruugey) [inline tools/build/tsconfig.json:8]: this can be rm'd

File: tools/test-harnesses/package.json [hunk 3]
```diff
@@ -13,6 +13,7 @@
     "access": "public"
   },
   "dependencies": {
-    "@testing-library/dom": "9.3.1"
+    "@testing-library/dom": "9.3.1",
+    "jsdom": "^24.0.0"
```
Comment (bruugey) [inline tools/test-harnesses/package.json:17]: this can be rm'd

File: packages/select/src/utils/getTestUtils/getTestUtils.ts [hunk 4]
```diff
@@ -18,7 +18,7 @@ export const getTestUtils = (
    * Queries the DOM for the element using the `data-lgid` data attribute.
    * Will throw if no element is found.
    */
-  const element = getByLgId(lgId);
+  const element = getByLgId!(lgId);
```
Comment (TheSonOfThomp) [inline packages/select/src/utils/getTestUtils/getTestUtils.ts:21]: Can we verify that this doesn't break anything on ssr?

File: tools/test-harnesses/src/utils/getDocument/index.ts [hunk 5]
```diff
@@ -5,7 +5,7 @@
 
 export function getDocument() {
   if (typeof window === 'undefined') {
-    throw new Error('Could not find default container');
+    return undefined;
```
Comment (TheSonOfThomp) [inline tools/test-harnesses/src/utils/getDocument/index.ts:8]: technically can just write `return`

File: tools/test-harnesses/src/utils/getQueries/index.ts [hunk 6]
```diff
@@ -5,17 +5,25 @@ import { getDocument } from '../getDocument';
...
-const {
-  getByTestId: getByLgId,
-  findByTestId: findByLgId,
-  queryByTestId: queryByLgId,
-} = queriesForElements;
-
-export { findByLgId, getByLgId, queryByLgId };
+let findByLgId;
+let getByLgId;
+let queryByLgId;
```
Comment (TheSonOfThomp) [inline tools/test-harnesses/src/utils/getQueries/index.ts:10]: Forking return/export values with a `let` declaration is a bit of a code-smell for me.

I would wrap this all in a function, something like this:
```typescript
function getBaseQueries() {
  const baseElement = getDocument();

  if (!baseElement) {
    return {
      findByLgId: undefined,
      getByLgId: undefined,
      queryByLgId: undefined,
    };
  }

  if (baseElement) {
    return getQueriesForElement(...);
  }
}

const queries = getBaseQueries();
export { queries };
``

Comment (TheSonOfThomp) [reply tools/test-harnesses/src/utils/getQueries/index.ts:10]: A benefit of this is that TS knows that if one query is undefined, they all are

Comment (stephl3) [reply tools/test-harnesses/src/utils/getQueries/index.ts]: updated!

**Review by TheSonOfThomp — APPROVED** [review]

**Pull request merged**